File: 프로그래머스/lv2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        Queue<Integer> q1 = new ArrayDeque<>();
        Queue<Integer> q2 = new ArrayDeque<>();
        long sum1 = 0;
        long sum2 = 0;
        for(int i = 0; i<queue1.length; i++){
            sum1 += queue1[i];
            sum2 += queue2[i];
            q1.offer(queue1[i]);
            q2.offer(queue2[i]);
        }
        
        int limit = queue1.length + 10;
        
        int p1 = 0;
        int p2 = 0;
        
        while (p1 < limit && p2 < limit){
            if (sum1 < sum2){
                int num = q2.poll();
                sum1 += num;
                sum2 -= num;
                q1.offer(num);
                p1++;
            }
            else if(sum1 > sum2){
                int num = q1.poll();
                sum1 -= num;
                sum2 += num;
                q2.offer(num);
                p2++;
            }
            else{
                return p1+p2;
            }
        }
        return -1;
        
    }
}

File: 프로그래머스/lv2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    q1 = deque()
    q2 = deque()
    sum1 = 0
    sum2 = 0
    for i in range(len(queue1)):
        sum1 += queue1[i]
        sum2 += queue2[i]
        q1.append(queue1[i])
        q2.append(queue2[i])

    limit = len(queue1) + 10

    p1 = 0
    p2 = 0

    while p1 < limit and p2 < limit:
        if sum1 < sum2:
            num = q2.popleft()
            sum1 += num
            sum2 -= num
            q1.append(num)
            p1 += 1
        elif sum1 > sum2:
            num = q1.popleft()
            sum1 -= num
            sum2 += num
            q2.append(num)
            p2 += 1
        else:
            return p1 + p2
    return -1

